async/abortable.ts

```diff
@@ -0,0 +1,72 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+import { deferred } from "./deferred.ts";
+
+export class AbortedError extends Error {
```

Comment:
I wonder if it makes sense to use DOMException with AbortError  instead of this custom error:

```ts
new DOMException(reason ? `Aborted: ${reason}` : "Aborted", "AbortError")
```

Reply:
Yes i think this would be preferable imo. Ideally it would be using error causes, but DOMException doesnt support that yet (https://github.com/whatwg/webidl/issues/969).

Reply:
Shall I rewrite it as `DOMException` with that?

Reply:
> Shall I rewrite it as DOMException with that?

I think that's more preferable (fetch API rejects with that error, for example). Let's do that